File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/bridge/solution/square_body.cpp
// square_body.cpp

#include "square_body.h"

square_body::square_body(double a_side)
{
	my_side = a_side;
}

double square_body::get_height() const
{
	return my_side;
}

double square_body::get_width() const
{
	return my_side;
}

bool square_body::is_square() const
{
	return true;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/bridge/solution/SquareBody.java
// SquareBody.java

class SquareBody extends RightAngledQuadrilateralBody
{
	public SquareBody(double aSide)
	{
		mySide = aSide;
	}

	public double getHeight()
	{
		return mySide;
	}

	public double getWidth()
	{
		return mySide;
	}

	public boolean isSquare()
	{
		return true;
	}

	private double mySide;
}
